HPCC-14433 Allow user <hpcc> to run hpcc-init and dafilesrv

xwang2713 opened the pull request:
<hpcc> user cannot run hpcc-init and dafilesrv since these script only allow root user to run.
But we used to allow <hpcc> user to start these services which is also helpful in non-root user privilege case.  
To fix it:
1) remove is_root in hpcc-init and dafilesrv
2) move creating /var/log/HPCCSystems to install-init script
3) remove "su <hpcc> -c " if current user is <hpcc> user to prevent prompts for password
4) Make sure debug log file owned by <hpcc> user

Signed-off-by: Xiaoming Wang xiaoming.wang@lexisnexi.com

@Michael-Gardner  please review

wangkx committed:
HPCC-12686 Use 3 enumerations in ws_dfu/ws_fs/ws_wu ecm files

1. in ws_dfu.ecm, change DFUArrayActionRequest.type to ESPenum;
2. in ws_fs.ecm, change DFUWorkunitsActionRequest.type to ESPenum;
3. in ws_workunits.ecm, change WUActionRequest.ActionType to ESPenum;
4. refactor the CFileSprayEx::onDFUWorkunitsAction().

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

ghalliday committed:
HPCC-12267 Fix already sorted to allow distribution to mismatch

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-12267 Test case for QUANTILE activity

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-12267 Add QUANTILE to the ecl parser

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-12267 Define code generator/engine interface for QUANTILE

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-12267 Add QUANTILE support to the code generator

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-12267 Initial implementation of roxie QUANTILE activity

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-12267 Finish roxie implementaton QUANTILE activity

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-13895 Fix problems with embedded datasets in nested rows

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

mckellyln committed:
HPCC-13451 Added config setting for Roxie multicast TTL

Changed to now set TTL for all MC sockets.
Default TTL if not set in xml is 1
If TTL is 0 then do not explicitly set TTL
to behave like before (using OS defaults)
Added multicastTTL="1" to default environment.xml

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

wangkx committed:
HPCC-13958 Return CanReplicate in WsDfu.DFUInfo

This fix checks and returns whether a logical file can
be replicated on a cluster or not. The information will
be used by new ECLWatch to show a Replicate button on
the Logical File Details page. The file can only be
replicated on a cluster if: 1. the cluster is a thor
cluster; 2. the cluster has multiple nodes; and 3. the
@replicateOutputs attribute is set to true.

Change based on review: 1. Cache node group info to a
hash table. 2. the default timeout value of the cache
can be modified using ConfigMgr.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

richardkchapman committed:
HPCC-14043 CWorkunitResumeHandler reads dali workunit branch directly

Refactor to use dali subscriptions whenever waiting for workunits, and for
Cassandra workunit repository to ensure that it triggers them when state or
action change.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-14031 Sasha accesses dali workunit tree directly

Minor changes following review.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-13899 Add support for graph progress in Cassandra

Previous changes for this had some issues in the dali connection caching.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7652 from richardkchapman/cassandra-sasha

HPCC-14031 Sasha accesses dali workunit tree directly

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7668 from richardkchapman/cassandra-nodestate

HPCC-13899 Add support for graph progress in Cassandra

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7664 from richardkchapman/cassandra-waitpaused

HPCC-14043 CWorkunitResumeHandler reads dali workunit branch directly

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-14048 Regression test apersistschedule1.ecl hangs in cassandra

Scheduled event count not being properly saved in workunits meant that the
scheduler was not working properly.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

RussWhitehead committed:
Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

RussWhitehead committed:
Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

RussWhitehead committed:
Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

richardkchapman committed:
Merge pull request #7666 from RussWhitehead/wuBlocked

HPCC-10811 ECL WU should indicate that an eclagent job is 'spraying'

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-14055 Error from regression suite test "readresult.ecl"

This highlighted a couple of regressions introduced into workunitservices by
prior fixes.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7670 from richardkchapman/cassandra-schedule

HPCC-14048 Regression test apersistschedule1.ecl hangs in cassandra

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com> 
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7671 from richardkchapman/cassandra-readresult

HPCC-14055 Error from regression suite test "readresult.ecl"

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-13150 Keydiff.ecl fails for multiPart=false

Ignore...

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-13874 Improve mySQL error messages

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-14083 Avoid early removal of spilling callback

When an output of a shared spilling stream is exhausted the
spill callback was removed, this commit leaves the callback in
place, but removes the write position callback, which is no
longer needed for the completed output.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #7633 from mckellyln/mcttl

HPCC-13451 Added config setting for Roxie multicast TTL

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-13526 Writing to file fails in roxie stand alone mode

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-12548 Roxie in memory indexes don't check for wrapping 2^32

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

mckellyln committed:
HPCC-14073 Removed unused jbroadcast code

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

richardkchapman committed:
Merge pull request #7692 from mckellyln/hpcc-14073

HPCC-14073 Removed unused jbroadcast code

Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7574 from jakesmith/hpcc-13545

HPCC-13545 Expose lock info to clients.

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7631 from wangkx/h13905

HPCC-13905 Return child dataset structure in DFUGetFileMetaData

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7693 from wangkx/h12686

HPCC-12686 Use 3 enumerations in ws_dfu/ws_fs/ws_wu ecm files

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7695 from jakesmith/hpcc-14083

HPCC-14083 Avoid early removal of spilling callback

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-13537 changes to thor status message

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

Michael-Gardner committed:
hpcc-13525 added whitelist file patterns for testing directory

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

AttilaVamos committed:
HPCC-13906 A reliable way to identify the type of file on a cluster

Store file type as "@kind" attribute in spray/copy.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

richardkchapman committed:
Merge branch 'candidate-5.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
hpcc-13997 Added red error message to cluster_script.py error output

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

mckellyln committed:
HPCC-14090 Suppress MP connection closed msg for 0 bytes sent

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

wangkx committed:
HPCC-14104 Check getWorkflowItems() return before using

The 6.0 ESP crashes when running ecl in ECLPLayground.
It happens when ECLPlayground calls WsWorkunits.WUINfo.
The getWorkflowItems() returns NULL when it is called in
WUInfo. WUINfo should check the return before using it.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

Michael-Gardner committed:
HPCC-13101 Added option(WITH_PLUGINS) to allow for disabling all plugins

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

Michael-Gardner committed:
HPCC-14113 Cleaned up unneeded files leftover in /tmp

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

wangkx committed:
HPCC-13547 Add WsSMC.LockQuery to report locks

This new method is used to retrieve locks with optional
filters: endpoint IP, duration, locked time, mode (READ,
WRITE, etc), and xpath.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

jamienoss committed:
HPCC-14116 Redis plugin: fail when attempting to initiate a persistent lock

Signed-off-by: James Noss <james.noss@lexisnexis.com>

richardkchapman committed:
Merge pull request #7720 from Michael-Gardner/hpcc-13997

hpcc-13997 Added red error message to cluster_script.py error output

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>

richardkchapman committed:
Merge pull request #7729 from jamienoss/issue14116-fail-when-initiating--persistent-locks

HPCC-14116 Redis plugin: fail when attempting to initiate a persistent lock

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-13980 Added configgen checks and validation checks for xml

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-5.4.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-14129 Clean up reporting output in hpcc-run.sh

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

wangkx committed:
HPCC-14097 Add multiple thor logs to ZAP report

The existing ZAP report can only include one thor log.
If a WU is in >1 thor logs (load balanced Thor), the
last thor log list in WU XML is copied into the ZAP
report. This fix adds multiple thor logs to ZAP report
if related.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

Michael-Gardner committed:
HPCC-14153 Modified ws_machineService and ws_machineServiceRexec to use CFRunSSH

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

RussWhitehead committed:
HPCC-14115 LDAP Enumerations need to support paging

LDAP imposes a limit of 1,000 on the number of entries that can be returned
from any search. Once there are over 1,000 matches to any search, an error
is returned and you cannot even get the first 1000.  This PR implements a
CPagedLDAPSearch class, which wraps code to perform an LDAP Paged Search,
and uses it in any ILDAPClient calls that could return this many entries.
It also implements a countEntries method, which is used in several places
within LDAP clients.  This includes:

retrieveUsers
getResources
getResourcesEx
getAllGroups
getGroupMembers
countEntries

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-5.4.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-14146 maxConnectTime ignored when clientSetRemoteFileTimeouts set

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
hpcc-14132 Fix ssh calls from scripts to include LogLevel=QUIET

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

richardkchapman committed:
Merge pull request #7721 from mckellyln/hpcc-14090

HPCC-14090 Suppress MP connection closed msg for 0 bytes sent

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7686 from richardkchapman/mysql-errors

HPCC-13874 Improve mySQL error messages

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #7744 from richardkchapman/maxConnectTime

HPCC-14146 maxConnectTime ignored when clientSetRemoteFileTimeouts set

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7741 from Michael-Gardner/hpcc-14129

HPCC-14129 Clean up reporting output in hpcc-run.sh

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7508 from ghalliday/issue13845

HPCC-13845 Improve the generated code for constant inline datasets

Reviewed-By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7742 from wangkx/h14097

HPCC-14097 Add multiple thor logs to ZAP report

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

mckellyln committed:
HPCC-14162 Release sock before continuing

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

ghalliday committed:
Merge pull request #7685 from richardkchapman/keydiff_meta

HPCC-13150 Keydiff.ecl fails for multiPart=false

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #7690 from richardkchapman/roxie-localwrite

HPCC-13526 Writing to file fails in roxie stand alone mode

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-12872 TRACE activity

Implement a new TRACE activity.

Syntax is:

myds := TRACE(ds [, traceOptions]);

The available options are:

 - Zero or more expressions, which act as a filter. Only rows matching the
   filter will be included in the tracing.
 - KEEP (n) indicating how many rows will be traced.
 - SKIP (n) indicating that n rows will be skipped before tracing starts.
 - SAMPLE (n) indicating that only every nth row is traced.
 - NAMED(string) providing the name for the rows in the tracing.

Tracing is output to the log file, in the form
 TRACE: <name><fieldname>value</fieldname>...</name>

Tracing is not output by default even if TRACE statements are present - only
if the workunit debug value traceEnabled is set or if the default platform
settings are changed to always output tracing. In Roxie you can also request
tracing on a deployed query by specifying traceEnabled=1 in the query XML. It
is therefore possible to leave trace statements in the ECL without any
detectable overhead until tracing is enabled.

It is also possible to override the default value for KEEP at a global,
per-workunit, or per-query level.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jamienoss committed:
HPCC-14053 Comments not always ignored when resolving macro parameter IDs

Signed-off-by: James Noss <james.noss@lexisnexis.com>

richardkchapman committed:
Merge pull request #7751 from mckellyln/hpcc-14162

HPCC-14162 Release sock before continuing

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

wangkx committed:
HPCC-13958 Revise based on code review

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

ghalliday committed:
Merge pull request #7758 from jamienoss/issue14053-ignore-comments-when-resolving-macro-parameter-UNKOWNIDs

HPCC-14053 Comments not always ignored when resolving macro parameter IDs

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #7565 from ghalliday/issue13895

HPCC-13895 Fix problems with embedded datasets in nested rows

Reviewed-By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7737 from Michael-Gardner/hpcc-13980

HPCC-13980 Added configgen checks and validation checks for xml

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7727 from Michael-Gardner/hpcc-14113

HPCC-14113 Cleaned up unneeded files leftover in /tmp

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7750 from Michael-Gardner/ws_machineService

HPCC-14153 Modified ESP machineService to use CFRunSSH

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

AttilaVamos committed:
HPCC-14181 'daliadmin validatestore' command generates core if 0
workunit associated file gathered.

Check the connect to "/GeneratedDlls" was succesfull.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

richardkchapman committed:
Merge pull request #7746 from Michael-Gardner/hpcc-14132

HPCC-14132 Fix ssh calls from scripts to include LogLevel=QUIET

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7765 from AttilaVamos/HPCC-14181-fix

HPCC-14181 'daliadmin validatestore' command generates core if 0 workunit associated file gathered.

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7759 from richardkchapman/trace

HPCC-12872 TRACE activity

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #7691 from richardkchapman/roxie-indexwrap

HPCC-12548 Roxie in memory indexes don't check for wrapping 2^32

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-12907 Roxie graph links with no records are not shown as "0"

Also refactored the code slightly to make it clearer how Roxi tracks other
stats, and to rationalize some stats between Roxie and Thor.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-8710 Multi slave in same process framework

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

Commit:
HPCC-14121 Multiple MatchBagOfWords matching requests (saltlib)

- Update lib_saltlib.ecllib

Signed-off-by: Edin Muharemagic <edin.muharemagic@lexisnexis.com>

Commit:
HPCC-14191 Configgen should report thormaster ports

Signed-off-by: Gleb Aronsky <gleb.aronsky@lexisnexis.com>

richardkchapman committed:
Merge pull request #7743 from RussWhitehead/ldapEnum

HPCC-14115 LDAP Enumerations need to support paging

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-13173 Constant folding should not call plugin functions

Only specifically whitelisted functions should be constant-folded.

Note that this does not prevent people from declaring their own services and
marking the functions in them as ,fold, so it does not close the security
loophole just yet - that will require some work on the code signing issue
HPCC-14185 - but this is a necessary step towards it (and a useful feature
in its own right).

Added ,NOFOLD attribute to override ,FOLD on individual functions within a
service.

Marked appropriate existing service functions foldable.

Refactored the external function call folding code to work around a compiler
bug on OSX.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-14160 Create cache variables for dependencies to be passed between projects

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

ghalliday committed:
Merge pull request #7768 from richardkchapman/roxie-stats2

HPCC-12907 Roxie graph links with no records are not shown as "0"

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #7654 from wangkx/h13958

HPCC-13958 Return CanReplicate in WsDfu.DFUInfo

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #7775 from garonsky/HPCC-14191_configgen_should_report_thormaster_port

HPCC-14191 Configgen should report thormaster ports

Reviewed-By: Michael Gardner <michael.gardner@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-14185 Support the option to sign code

Work in progress. Support separate permissions for PIPE, embedc++, and FOLD
depending on whether they are in signed code or not.

Actual signing bit remains TBD.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-12943 Added LD_PRELOAD export to hpcc_setenv

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

Michael-Gardner committed:
HPCC-14160 Added macro for dependencies

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

Michael-Gardner committed:
HPCC-14160 Changed around the order of the macro parameters

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-5.4.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.2'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
hpcc-14160 Cache variable set differently in platform

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

Michael-Gardner committed:
HPCC-14160 Made safe for reconfigure of cmake

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

wangkx committed:
HPCC-14205 Add paging functions to ws_access

Four new APIs are added: UserQuery, GroupQuery, and
ResourceQuery, and GroupMemberQuery.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

RussWhitehead committed:
HPCC-14220 Improve LDAP Paged Search memory management

The CPagedLDAPSearch::requestNextPage is somewhat complex with respect to
managing memory returned from several LDAP calls. Instead of remembering
to free the memory on exit, on error, and on exceptions, better to encapsulate
the pointers in a local class that deletes the memory when destructed

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Michael-Gardner committed:
HPCC-14122 Adding jvmoptions w/ -UsePerfData to environment.conf

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

AttilaVamos committed:
HPCC-14223 Reword Regression Test Engine retry attempt message.

Reword messages.

Remove hardwired timeout constants.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

ghalliday committed:
HPCC-14224 Adding missing entries from new trace activity

Would cause syntax errors to generate internal asserts

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #7798 from ghalliday/issue14224

HPCC-14224 Adding missing entries from new trace activity

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7797 from AttilaVamos/HPCC-14223-improvement

HPCC-14223 Reword Regression Test Engine retry attempt message.

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7792 from RussWhitehead/ldapMem

HPCC-14220 Improve LDAP Paged Search memory management

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7795 from Michael-Gardner/hpcc-14122

HPCC-14122 Adding jvmoptions w/ -UsePerfData to environment.conf

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7708 from Michael-Gardner/hpcc-13537

HPCC-13537 changes to thor status message

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-14150 Add feature to envgen to set any xpath attribute value

Signed-off-by: Gleb Aronsky <gleb.aronsky@lexisnexis.com>

richardkchapman committed:
Merge pull request #7786 from Michael-Gardner/hpcc-14160

HPCC-14160 Cache package dependencies

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7723 from AttilaVamos/HPCC-13906-improvement

HPCC-13906 A reliable way to identify the type of file on a cluster

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7800 from garonsky/HPCC-14150_improve_envgen

HPCC-14150 Add feature to envgen to set any xpath attribute value

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-14234 WHEN subgraph execution race condition fix

A race condition during execution of dependent WHEN subgraphs
can cause Thor to stall. If the slave begins to execute before
the master graph has triggered the completio of the parent graph,
the slave graph executor would queue the pending graph and never
pull it from the queue.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #7722 from wangkx/h14104

HPCC-14104 Check getWorkflowItems() return before using

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-14242 Constant-folding of external function calls could give wrong answer

Revealed only after refactoring the code in foldExternalCall (earlier in this
branch), the asm sections marshalling parameters were not correctly indicating
their clobber lists, which with some variants of gcc and in release-mode only
could result in the folded result being incorrect.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7802 from jakesmith/hpcc-14234

HPCC-14234 WHEN subgraph execution race condition fix

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #7793 from Michael-Gardner/hpcc-12943

HPCC-12943 Added LD_PRELOAD export to hpcc_setenv

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7738 from Michael-Gardner/hpcc-13101

HPCC-13101 Added option(WITH_PLUGINS) to allow for disabling all plugins

Reviewed-By: Chris Lo <christopher.lo@lexisnexis.com> 
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-14243 Add method to retrieve cluster group

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

Michael-Gardner committed:
HPCC-13525 Adding testing/regress/ecl to whitelist

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-5.4.4'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-14244 Enable ability to create/use multiple MP servers

Add ability to create multiple MP servers (each on different port).

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

Michael-Gardner committed:
HPCC-14245 Changed variable named java to JNI_PATH

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

afishbeck committed:
HPCC-14246 Allow DynamicESDL service methods to be written in JAVA

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

richardkchapman committed:
Merge pull request #7807 from Michael-Gardner/hpcc-14245

HPCC-14245 Changed variable named java to JNI_PATH

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7776 from richardkchapman/fold-restricted

HPCC-13173 Constant folding should not call plugin functions

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #7774 from emuharemagic/hpcc14121

HPCC-14121 Multiple MatchBagOfWords matching requests (saltlib)

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #7718 from Michael-Gardner/hpcc-13525

HPCC-13525 Added whitelist file patterns for testing directory

Reviewed-By: Attila Vamos <attila.vamos@lexisnexis.com>
Reviewed-By: Gavin Halliday <gavin.halliday@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-14243 A couple of minor changes following review

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-5.4.4'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-14252 Clear up a few minor MP leaks.

Some spurious Linked vs Owned dealing with channels, meant they
were leaked and not cleared up on exit.
Has little impact since they are created once per destination,
and persist until end.

Also ensure worldComm is cleared up on module exit.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

afishbeck committed:
HPCC-14246 Update based on review

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

richardkchapman committed:
Merge pull request #7812 from jakesmith/hpcc-14252

HPCC-14252 Clear up a few minor MP leaks.

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7805 from jakesmith/hpcc-14243

HPCC-14243 Add method to retrieve cluster group

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Preparation for 6.0.0-beta1 release

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7814 from richardkchapman/beta-maturity

Preparation for 6.0.0-beta1 release

wangkx committed:
HPCC-13547 Revise based on review

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

wangkx committed:
HPCC-13547 Revise based on review

And throw an exception if lock mode filter is invalid.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

richardkchapman committed:
HPCC-14185 Support the option to sign code

Use gpg to check signatures

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-14185 Support the option to sign code

Remove the original signing checks now that we do it in lexer.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7809 from afishbeck/espJavaServicesXY

HPCC-14246 Allow DynamicESDL service methods to be written in JAVA

Reviewed-By: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-14212 Previous fix still left a small window for deadlock.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #7816 from jakesmith/hpcc-14212-v3

HPCC-14212 Previous fix still left a small window for deadlock.

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Community Edition 5.4.2-rc5 Release Candidate 5

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.2' into candidate-5.4.4

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.4'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7815 from richardkchapman/signed-allow

HPCC-14185 Support the option to sign code

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.4'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

afishbeck committed:
HPCC-14256 Add DynamicESDL java services support for simple arrays

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

richardkchapman committed:
HPCC-14185 Support the option to sign code

Missing edit in prior fix caused pipe8 regression test to fail.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-12267 Temporarily restrict quantile regression tests to roxie

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
Merge pull request #7819 from richardkchapman/signed-allow2

HPCC-14185 Support the option to sign code

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #7808 from wangkx/h14148

HPCC-14205 Add paging functions to ws_access

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7785 from wangkx/h13537

HPCC-13547 Add WsSMC.LockQuery to report locks

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jamienoss committed:
HPCC-14109 Redis plugin - extend timeouts to duration of plugin calls

This also changes the type for database and refactors error messaging

Signed-off-by: James Noss <james.noss@lexisnexis.com>

ghalliday committed:
HPCC-14263 Add extra fold attributes where necessary

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
HPCC-14246 Allow DynamicESDL service methods to be written in JAVA

Fix some build issues.

Load java plugin dynamically so that jvm is only loaded when the feature is
used.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7822 from ghalliday/issue14263

HPCC-14263 Add extra fold attributes where necessary

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-14267 Make trace.ecl test run on single node

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #7823 from jakesmith/hpcc-14267

HPCC-14267 Make trace.ecl test run on single node

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7796 from jamienoss/issue14109-Redis-plugin-change-timeout-functionality-alpha-rebase

HPCC-14109 Redis plugin - extend timeouts to duration of plugin calls

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

afishbeck committed:
HPCC-14256 Update based on review

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

ghalliday committed:
Merge pull request #7821 from richardkchapman/esdl-compile-errors

HPCC-14246 Allow DynamicESDL service methods to be written in JAVA

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-14272 Compile errors from latest XCode

Highlighting a probably (minor) bug in the code.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7818 from afishbeck/espJavaServiceSimpleArray

HPCC-14256 Add DynamicESDL java services support for simple arrays

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7806 from jakesmith/hpcc-14244

HPCC-14244 Enable ability to create/use multiple MP servers

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #7813 from jakesmith/hpcc-8710

HPCC-8710 Multi slave in same process framework

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #7820 from ghalliday/issue12267

HPCC-12267 Initial implementation of QUANTILE activity in roxie

Reviewed-By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Community Edition 6.0.0-betarc1 Beta 1 Release Candidate 1

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Split off 6.0.0 branch

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

AttilaVamos committed:
HPCC-14266 TargetIP being ignored and assumed to be same as engine IP's.

Fix "TargetIP" override by default value from ecl-test.json problem.

Remove "IpAddress" array from ecl-test.json configuration file.

Add "espIp" parameter to ecl-test.json file.

Modify Regression Test Engine to use espIp value istead of the target
engine related IP from "IpAddress" array.

Update README.rst

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

richardkchapman committed:
Merge pull request #7825 from richardkchapman/osx-picky

HPCC-14272 Compile errors from latest XCode

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>

jamienoss committed:
HPCC-14286 Redis Plugin - add publish and subscribe functions

Signed-off-by: James Noss <james.noss@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-5.4.4' into candidate-6.0.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'master' into candidate-6.0.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

wangkx committed:
HPCC-13547 Initialize ttLTHigh and ttLTLow (compile warnings)

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

RussWhitehead committed:
HPCC-14293 Client Tools build break on LDAP component

LDAP API Signatures differ between OpenLDAP and WinLDAP. Also exported
LDAPConnection functions need correct linkage defined

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

richardkchapman committed:
Merge pull request #7832 from wangkx/h13547_2

HPCC-13547 Initialize ttLTHigh and ttLTLow (compile warnings)

Reviewed By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Community Edition 6.0.0-beta1 Beta 1

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

RussWhitehead committed:
HPCC-14291 LDAP build fails on Centos5

Centos5 LDAP support is very downlevel, so the new API added to 6.0 to support
LDAP paging is not included in the OpenLDAP file ldap.h.  This PR checks for the
   #define LDAP_API_FEATURE_PAGED_RESULTS
that was added with the new API, and if does not exist falls back to the old
way of querying attributes

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Commit:
HPCC-13195 Docs: Change term Farmer to Roxie server

Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

Commit:
HPCC-14247 Document FOLD / NOFOLD options for SERVICE

Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

Commit:
HPCC-14247 Add index terms to doc

Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

jakesmith committed:
HPCC-14323 Bad type definition on processPerSlave

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

Commit:
HPCC-14226 Document range of values for RANDOM

Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

richardkchapman committed:
HPCC-14305 6.0.0-beta1 does not build on centos 5

CPU_COUNT macro not available on earlier versions of glibc. Code stolen from
jlib.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-14380 Fix various c++11 compile warnings

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
Merge pull request #7835 from RussWhitehead/fixCentos5

HPCC-14291 LDAP build fails on Centos5

Reviewed-By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7837 from JamesDeFabia/13195Farmer

HPCC-13195 Docs: Change term Farmer to Roxie server

Reviewed-By: Greg Panagiotatos <Greg.Panagiotatos@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7833 from RussWhitehead/ldapBreak

HPCC-14293 Client Tools build break on LDAP component

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.4' into candidate-6.0.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'master' into candidate-6.0.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-6.0.0'

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-14380 Fix warnings and errors on windows builds

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

wangkx committed:
HPCC-14342 Move #define ... __declspec from ipp to hpp

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

Commit:
HPCC-14326 Add delete flag to STD.File.RemoveOwnedSubFiles

Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

richardkchapman committed:
Merge pull request #7838 from JamesDeFabia/14247FoldNofold

HPCC-14247 Document FOLD / NOFOLD options for SERVICE

Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

jamienoss committed:
HPCC-14357 Lang Ref Man missing ',' in WHEN syntax

Signed-off-by: James Noss <james.noss@lexisnexis.com>

richardkchapman committed:
Merge pull request #7827 from AttilaVamos/HPCC-14266-fix

HPCC-14266 TargetIP being ignored and assumed to be same as engine IP's.

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-14167 Improve docs for -I parameter in ecl cli

Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

richardkchapman committed:
Merge pull request #7851 from jamienoss/issue14357-lan-ref-missing-comma-WHEN-syntax

HPCC-14357 Lang Ref Man missing ',' in WHEN syntax

Reviewed-By: Jim DeFabia <james.defabia@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
HPCC-14385 Minor fixes in preparation for HPCC-9318

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

Michael-Gardner committed:
HPCC-14333 Modified CFRunSSH code in ws_machineService.cpp

Modified user.str() and password.str() to use null strings so that we
force the user in ssh to be the same as the process which is calling the
ssh session.

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

richardkchapman committed:
Merge pull request #7849 from wangkx/h14342

HPCC-14342 Move #define ... __declspec from ipp to hpp

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7840 from jakesmith/hpcc-14323

HPCC-14323 Bad type definition on processPerSlave

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.4' into candidate-6.0.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'master' into candidate-6.0.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7854 from ghalliday/issue14380

HPCC-14380 Fix various c++11 compile warnings

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge remote-tracking branch 'gavin/issue14385' into candidate-6.0.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

wangkx committed:
HPCC-14383 Change WUListQueries filters to be case insensitive

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

richardkchapman committed:
HPCC-14388 Force full warnings as errors for embedded C++

Note that as well as enabling all warnings as errors for user code, this also
disables all warnings for generated code - otherwise as soon as there is an
error (from the user code) they get swamped with all the warnings from the
generated code.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-14409 Introduce a row manager per slave channel.

Avoids heavy (CPU bound) contention inside single RM,
and voids slave channels starving each other of memory

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #7864 from wangkx/h14383

HPCC-14383 Change WUListQueries filters to be case insensitive

Reviewed By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7844 from richardkchapman/centos5-affinity

HPCC-14305 6.0.0-beta1 does not build on centos 5

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #7841 from JamesDeFabia/HPCC-14226DocumentRANDOMrange

HPCC-14226 Document range of values for RANDOM

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

RussWhitehead committed:
HPCC-14382 LDAP build break on Windows

WinLDAP defines ldap_init differently than the OpenLDAP call ldap_initialize,
which causes a platform build break in Windows. This PR corrects that problem,
and adds the export of the verifyServerCert method

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

Commit:
HPCC-14167 Edits based upon review

Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

jamienoss committed:
HPCC-14401 Redis Plugin - catch all NULL replies

This also defers the error checking of the reply/connection of the initial
sub request until actually needed - if at all.

Signed-off-by: James Noss <james.noss@lexisnexis.com>

jakesmith committed:
HPCC-14409 add a activity getRowAllocator queryRowManager

cosmetic change.

Add CActivityBase::getRowAllocator and a
CActivityBase::queryRowManager, for slightly more readibility in
activity code.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

ghalliday committed:
Merge pull request #7879 from jamienoss/issue14401-redis-plugin-ensure-null-replies-are-caught

HPCC-14401 Redis Plugin - catch all NULL replies

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #7853 from jamienoss/issue14286-redis-plugin-add-independent-pub-sub-functionality-with-db

HPCC-14286 Redis Plugin - add publish and subscribe functions

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

RussWhitehead committed:
Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

richardkchapman committed:
Merge pull request #7852 from JamesDeFabia/14167DocsImportinECL_cli

HPCC-14167 Improve docs for -I parameter in ecl cli

Reviewed By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

RussWhitehead committed:
Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

richardkchapman committed:
Community Edition 5.4.4-rc1 Release Candidate 1

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.4' into candidate-6.0.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.4' into candidate-6.0.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7850 from JamesDeFabia/14326DeleteFlatOnRemoveOwnedSubFiles

HPCC-14326 Add delete flag to STD.File.RemoveOwnedSubFiles

Reviewed-By: Attila Vamos <attila.vamos@lexisnexis.com>
Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

jamienoss committed:
Revert "HPCC-14286 Redis Plugin - add publish and subscribe functions"

ghalliday committed:
Merge pull request #7888 from jamienoss/revert-7853-issue14286-redis-plugin-add-independent-pub-sub-functionality-with-db

Revert "HPCC-14286 Redis Plugin - add publish and subscribe functions"

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #7876 from RussWhitehead/WinLDAPFix

HPCC-14382 LDAP build break on Windows

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

jamienoss committed:
HPCC-14286 Redis Plugin - Add independent pub/sub functions

Signed-off-by: James Noss <james.noss@lexisnexis.com>

jamienoss committed:
Rearrange tests

Signed-off-by: James Noss <james.noss@lexisnexis.com>

jamienoss committed:
Manually merge HPCC-14418 #7887

Signed-off-by: James Noss <james.noss@lexisnexis.com>

ghalliday committed:
Merge pull request #7890 from jamienoss/issue14286-redis-plugin-add-independent-pub-sub-functionality-with-db-no-cachedSubConnection-c

HPCC-14286 Redis Plugin - Add independent pub/sub functions

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #7881 from jakesmith/hpcc-14409

HPCC-14409 Introduce a row manager per slave channel.

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #7861 from Michael-Gardner/HPCC-14333

HPCC-14333 Modified CFRunSSH code in ws_machineService.cpp

Reviewed-By: Mark Kelly <mark.kelly@lexisnexis.com>
Reviewed-By: Kunal Aswani <kunal.aswani@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_5.4.4-rc1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

Michael-Gardner commented:
@xwang2713 I can't get this to work on a fresh installation.  The HPCCSystems directories in /var/lib, /var/run, /var/lock, /var/log need to be laid down w/ a chown hpcc:hpcc beforehand.

xwang2713 commented:
Add "createRuntime" to create create runtime directories and chown to <hpcc> user

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_5.4.4-rc1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

ghalliday committed:
Merge pull request #7869 from richardkchapman/cpperrors

HPCC-14388 Force full warnings as errors for embedded C++

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

jakesmith committed:
HPCC-14388 Linux build error (missing " terminator on #pragma)

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #7901 from jakesmith/hpcc-14388

HPCC-14388 Linux build error (missing " terminator on #pragma)

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner commented on initfiles/bash/etc/init.d/install-init.in:
```diff
@@ -300,3 +300,13 @@ fi
+fi
+[ -e  ${LOG_DIR}/hpcc-init.debug ] && chown ${user}:${user} ${LOG_DIR}/hpcc-init.debug
+
+createRuntime
```
Ming I think if you move this up to line 304, you can them remove the LOG_DIR block.  Im pretty sure it's redundant in the createRuntime function.  Other than that, this looks solid.  Tested it and it works fine.

xwang2713 replied:
Actually "$compName", referenced by createRuntime, is not defined so install may throw error unless modifying createRuntime function. As user may change HPCC cluster topology I wonder if it is worthy we process the runtime setting during install.

Michael-Gardner commented:
yeah the empty $compName could be an issue, but since we have it unset it should evaluate out to "".  I've checked createRuntime and it doesn't look like there should be an issue with an empty compName.  Not 100% certain though.  It did work in my tests.

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_5.4.4-rc1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman committed:
Merge branch 'candidate-5.4.4' into candidate-6.0.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

xwang2713 commented:
@Michael-Gardner please help to review again for the change I made.

richardkchapman commented:
This has missed the boat for 5.4.4 - will need to be rebased for 5.4.6 or 6.0.0 as appropriate

xwang2713 commented:
OK. I will rebase for 6.0.0

xwang2713 committed:
HPCC-14433 Allow <hpcc> user to run hpcc-init and dafilesrv

remove is_root in hpcc-init and dafilesrv
move creating /var/log/HPCCSystems to install-init script
remove "su <hpcc> -c " if current user is <hpcc> user
Make sure debug log file owned by <hpcc> user
Create runtime directories and chown to <hpcc> user
Add handle empty $compName in createRuntime and remove redundant code
for generating log directores

HPCCSmoketest commented:
Automated Smoketest Build: failed
Number of Error(s): 4
/mnt/disk1/home/vamosax/smoketest/smoketest-7899/HPCC-Platform/plugins/redis/redis.cpp:382:27: error: ‘class RedisPlugin::Connection’ has no member named ‘updateTimeout’
make[2]: *** [plugins/redis/CMakeFiles/redis.dir/redis.cpp.o] Error 1
make[1]: *** [plugins/redis/CMakeFiles/redis.dir/all] Error 2
make: *** [all] Error 2

xwang2713 commented:
Will re-open against HPCC branch candidate-6.0.0

xwang2713 commented:
Will re-open for HPCC candidate-6.0.0
